File: 978-valid-mountain-array/valid-mountain-array.java
class Solution {
    public boolean validMountainArray(int[] nums) {
        if(nums.length<3) return false;
        int low =0;
        int high = nums.length-1;
        int n = nums.length-1;
        if(nums[0]>nums[1]) return false;
        if(nums[high]>nums[high-1]) return false;
        while(low<n-1)
        {
            if(nums[low+1]>nums[low]) low++;
            else if(nums[low]==nums[low+1]) return false;
            else break;
        }
        while(high>0)
        {
            if(nums[high-1]>nums[high]) high--;
            else if(nums[high]==nums[high-1]) return false;
            else break;
        }
        System.out.println(low+ " "+high);
        if(low==high) return true;
        else return false;
    }
}

File: 978-valid-mountain-array/valid-mountain-array.cpp
class Solution {
public:
    bool validMountainArray(vector<int>& arr) {

        int n = arr.size()-1;
        if(arr.size()<3) return false;
        if(arr[0]>=arr[1] || arr[n]>=arr[n-1]) return false;
        int i=1;
        while(i<n)
        {
            if(arr[i]>arr[i-1]) i++;
            else if(arr[i]==arr[i-1]) return false;
            else break; 
        }
        i = i-1;
        while(i<n-1)
        {
            if(arr[i]>arr[i+1]) i++;
            else if(arr[i]==arr[i+1]) return false;
            else break;
        }
        cout<<i<<endl;
        return (i==n-1)?true:false;
    }
};